ts/src/modetrade.ts

```diff
@@ -0,0 +1,2874 @@
+     * @param {boolean} [params.paginate] set to true if you want to fetch trades with pagination
+     * @param {int} params.until timestamp in ms of the latest trade to fetch
+     * @returns {Trade[]} a list of [trade structures]{@link https://docs.ccxt.com/#/?id=trade-structure}
+     */
+    async fetchMyTrades (symbol: Str = undefined, since: Int = undefined, limit: Int = undefined, params = {}) {
+        await this.loadMarkets ();
+        let paginate = false;
+        [ paginate, params ] = this.handleOptionAndParams (params, 'fetchMyTrades', 'paginate');
+        if (paginate) {
+            return await this.fetchPaginatedCallIncremental ('fetchMyTrades', symbol, since, limit, params, 'page', 500) as Trade[];
+        }
+        let request: Dict = {};
+        let market: Market = undefined;
+        if (symbol !== undefined) {
+            market = this.market (symbol);
+            request['symbol'] = market['id'];
+        }
+        if (since !== undefined) {
+            request['start_t'] = since;
+        }
+        if (limit !== undefined) {
+            request['size'] = limit;
+        } else {
+            request['size'] = 500;
+        }
+        [ request, params ] = this.handleUntilOption ('end_t', request, params);
+        const response = await this.v1PrivateGetTrades (this.extend (request, params));
+        //
+        // {
+        //     "success": true,
+        //     "timestamp": 1702989203989,
+        //     "data": {
+        //       "meta": {
+        //         "total": 9,
+        //         "records_per_page": 25,
+        //         "current_page": 1
+        //       },
+        //       "rows": [{
+        //         "id": 2,
+        //         "symbol": "PERP_BTC_USDC",
+        //         "fee": 0.0001,
+        //         "fee_asset": "USDC",
+        //         "side": "BUY",
+        //         "order_id": 1,
+        //         "executed_price": 123,
+        //         "executed_quantity": 0.05,
+        //         "executed_timestamp": 1567382401000,
+        //         "is_maker": 1,
+        //         "realized_pnl": 123
+        //       }]
+        //     }
+        // }
+        //
+        const data = this.safeDict (response, 'data', {});
+        const trades = this.safeList (data, 'rows', []);
+        return this.parseTrades (trades, market, since, limit, params);
+    }
+
+    parseBalance (response): Balances {
+        const result: Dict = {
+            'info': response,
+        };
+        const balances = this.safeList (response, 'holding', []);
+        for (let i = 0; i < balances.length; i++) {
+            const balance = balances[i];
+            const code = this.safeCurrencyCode (this.safeString (balance, 'token'));
+            const account = this.account ();
+            account['total'] = this.safeString (balance, 'holding');
+            account['frozen'] = this.safeString (balance, 'frozen');
```

Comment:
got an error on freqtrade: "Available balance (0 USDC) is lower than stake amount (15 USDC)".

Balances: {'USDC': {'free': None, 'used': None, 'total': 45.0, 'frozen': '0.0'}

I think here we need to parse `free` and `used` balances

```suggestion
            account['frozen'] = this.safeString (balance, 'frozen');
            account['free'] = this.safeString (balance, 'free');
            account['used'] = this.safeString (balance, 'used');
```